File: codes/gcd.cpp
#include <iostream>
using namespace std;

// GCD function
int gcd(int A, int B) {
  // Inputs: two integers A and B
  // Output: the GCD of A and B

  A = abs(A);
  B = abs(B);

  while(B != 0) {
    if (A > B) A = A - B;
    else B = B - A;
  }
  return A;
}


int main() {
  int A, B;

  // input two integers
  cout << "Enter 1st integer A: ";
  cin >> A;

  cout << "Enter 2nd integer B: ";
  cin >> B;

  int result = gcd(A,B);
  cout << "The GCD: " << result << endl;

}

File: codes/gcd.py
# GCD Function
def gcd(A,B):
  A = abs(A)
  B = abs(B)
  while B != 0:
    if A > B:
      A = A - B
    else:
      B = B - A
  return A

if __name__ == "__main__":
  # input two integers
  A = int(input("Enter 1st integer A: "))
  B = int(input("Enter 2nd integer B: "))
  print("The GCD: ", gcd(A,B))